Decred Wallet/Features/Transactions/TransactionTableViewCell.swift

```diff
@@ -0,0 +1,115 @@
+        self.voteRewardLabel.isHidden = true //TODO: set value
+
+        if transaction.type == DcrlibwalletTxTypeRegular {
+            self.displayRegularTxInfo(transaction)
+        } else if transaction.type == DcrlibwalletTxTypeVote {
+            self.displayTicketPurchaseInfo(transaction)
+        } else if transaction.type == DcrlibwalletTxTypeTicketPurchase {
+            self.displayVoteTxInfo(transaction)
+        }
+    }
+    
+    func displayRegularTxInfo(_ transaction: Transaction) {
+        if transaction.direction == DcrlibwalletTxDirectionSent {
+            let attributedString = NSMutableAttributedString(string: "-")
+            attributedString.append(Utils.getAttributedString(str: transaction.dcrAmount.round(8).description, siz: 13.0, TexthexColor: UIColor.appColors.darkBlue))
+            self.txAmountOrTicketStatusLabel.attributedText = attributedString
+            self.txTypeIconImageView?.image = UIImage(named: "ic_send")
+        } else if transaction.direction == DcrlibwalletTxDirectionReceived {
+            let attributedString = NSMutableAttributedString(string: " ")
+            attributedString.append(Utils.getAttributedString(str: transaction.dcrAmount.round(8).description, siz: 13.0, TexthexColor: UIColor.appColors.darkBlue))
+            self.txAmountOrTicketStatusLabel.attributedText = attributedString
+            self.txTypeIconImageView?.image = UIImage(named: "ic_receive")
+        } else if transaction.direction == DcrlibwalletTxDirectionTransferred {
+            let attributedString = NSMutableAttributedString(string: " ")
+            attributedString.append(Utils.getAttributedString(str: transaction.dcrAmount.round(8).description, siz: 13.0, TexthexColor: UIColor.appColors.darkBlue))
+            self.txAmountOrTicketStatusLabel.attributedText = attributedString
+            self.txTypeIconImageView?.image = UIImage(named: "ic_fee")
+        }
+    }
+    
+    func displayTicketPurchaseInfo(_ transaction: Transaction) {
+        let ageInDays = Date(timeIntervalSince1970: TimeInterval(transaction.timestamp)).daysFromNow
+
+        self.txAmountOrTicketStatusLabel.text = " \(LocalizedStrings.voted)"
+        self.txTypeIconImageView?.image = UIImage(named: "ic_ticketVoted")
+
+        let attributedString = NSMutableAttributedString(string: " ")
+        attributedString.append(Utils.getAttributedString(str: transaction.dcrAmount.round(8).description, siz: 11.0, TexthexColor: UIColor.appColors.lightBluishGray))
+        self.stakingTxAmountLabel.attributedText = attributedString
+        self.daysCounterLabel.text = String(format: LocalizedStrings.days, -ageInDays)
+    }
+    
+    func displayVoteTxInfo(_ transaction: Transaction) {
+        let ageInDays = Date(timeIntervalSince1970: TimeInterval(transaction.timestamp)).daysFromNow
+
+        let requireConfirmation = Settings.spendUnconfirmed ? 0 : 2
+        let txConfirmations = transaction.confirmations
```

Comment:
Move these below to just before the if statement where `requireConfirmation` is used.